index.js

```diff
@@ -134,36 +146,43 @@ exports.registerTriggeredMethods = (componentName, stateMachineName, triggeredMe
     }
 }
 
-exports.startEventQueue = (configuration) => {
-    console.log('Registered triggered methods: ', triggeredMethods);
-
-    const installEventQueue = () => {
-        setInterval(eventQueue, 1000);
-        console.log('Waiting for tasks...');
-    };
-
+function updateConfiguration(configuration, callback) {
     if (configuration) {
         const postConfigurationAction = () => postConfiguration(configuration, configurationCallback);
-        const retryIntervalSeconds = 5;
         let configurationTimerID = null;
         const configurationCallback = (error) => {
             if (error) {
                 console.error('Error updating configuration: ', error);
-                console.error('Retrying in ', retryIntervalSeconds, 's...');
-                configurationTimerID = setTimeout(postConfigurationAction, retryIntervalSeconds * 1000);
+                callback(error, null);
                 return;
             }
-
-            clearTimeout(configurationTimerID);
             console.log('Configuraton successfuly updated!');
-            installEventQueue();
+            callback(null, true);
         };
 
         postConfigurationAction();
-        configurationTimerID = setTimeout(postConfigurationAction, retryIntervalSeconds * 1000);
+        installEventQueue(callback)
     } else {
-        installEventQueue();
+        installEventQueue(callback)
     }
+}
+
+function installEventQueue(callback) {
+    setInterval(eventQueue, 1000);
+    console.log('Waiting for tasks...');
+    callback(null, true);

```

Comment:
`callback && callback(null, true)`

the callback should be optional!

Reply:
The callback is actually optional. If callback is undefined, callback(null, true) is not executed

Reply:
hmmm no... if callback is undefined we get an error

https://jsbin.com/sixugojolu/edit?html,console,output

Reply:
ok